PyTorch/classification/train_classification.py

```diff
@@ -38,55 +50,82 @@ def train(dataloader, model, device, loss, learning_rate, momentum, weight_decay
             with profiler.profile(record_shapes=True, with_stack=True, profile_memory=True) as prof:
                 with profiler.record_function("model_inference"):
                     # Compute loss and perform backpropagation
-                    batch_loss = loss(model(X), y)
+                    if (model_str == 'inception_v3'):
+                        pred, _ = model(X)
```

Comment:
looks like inception_v3 and googlenet have more than one return param - instead of special casing based on the model name, should we special case based on the fact that it has more than one param? suppose a new model is added to the script that also has two return values - how would that be handled here?

Reply:
inception_v3 and googlenet are the special cases here.
Yeah, we could count the number the output instead of the model name, but the the ways how loss is computed based on the outputs are different for different models as you can at line 74 and 80. So anyway we have to know which model the script is training and compute the loss accordingly.